postgis/postgis.sql.in

```diff
@@ -3476,12 +3476,19 @@ CREATE OR REPLACE FUNCTION ST_OffsetCurve(line geometry, distance float8, params
        COST 1; -- reset cost, see #3675
 
 -- Availability: 2.3.0
-CREATE OR REPLACE FUNCTION ST_GeneratePoints(area geometry, npoints numeric)
+CREATE OR REPLACE FUNCTION ST_GeneratePoints(area geometry, npoints integer)
        RETURNS geometry
        AS 'MODULE_PATHNAME','ST_GeneratePoints'
-       LANGUAGE 'c' IMMUTABLE STRICT _PARALLEL
+       LANGUAGE 'c' VOLATILE STRICT _PARALLEL_RESTRICTED
```

Comment:
does it have to be parallel restricted? if it's volatile and resets seed on start why not start it in parallel worker too?

Reply:
Postgres has set seed() as a separate call from random(), that's why they can't migrate random() call into parallel worker. Here it probably not applies.

Reply:
As mentioned in [trac #4299](https://trac.osgeo.org/postgis/ticket/4299), the [docs for parallel safety](https://www.postgresql.org/docs/10/parallel-safety.html) say "`setseed` and `random` are parallel restricted", so I've applied this to this function too. However, I don't know much about the parallel features of postgres, so am open to modifying this aspect.